test/jdk/com/sun/jdi/ThreadMemoryLeakTest.java

```diff
@@ -58,14 +58,23 @@ public static void main(String[] args) throws InterruptedException {
         while (System.currentTimeMillis() - startTime < 100 * 1000) {
             iterations++;
             semaphore.acquire();
-            Executors.defaultThreadFactory().newThread(() -> {
+            TestScaffold.newThread(() -> {
                     adder.increment();
                     long sum = adder.sum();
                     if ((sum % 1000) == 0) {
                         System.out.println("Progress: " + sum);
                     }
                     try {
-                        Thread.sleep(50);
+                        String mainWrapper = System.getProperty("main.wrapper");
+                        if ("Virtual".equals(mainWrapper)) {
+                            // Virtual thread creation tends to overwhelm the debugger,
+                            // leading to high memory use for all the unprocessed events
+                            // that get queued up, so we need to slow it down a bit more
+                            // than we do for platform threads to avoid getting OOME.
+                            Thread.sleep(100);
```

Comment:
I wonder if this time for sleep can still be not enough.

Reply:
I did a lot of testing on all platforms, including with product builds, but yes, it is possible that on some some platforms with some flags it might not be enough. I guess more testing will tell. Adjustments might be necessary. It is important not too slow things down too much, or it's possible that if there is a memory leak, the test won't catch it because the leak is not fast enough. With the current sleep values, throughput for virtual threads is still about 2x what it is for platform threads, so right now I'm not worried about it having been slowed down too much.

Reply:
small nit. shorter to use:
long timeToSleep = "Virtual".equals(mainWrapper) ? 100 : 50;

Reply:
Okay.